helpers/announce.js

```diff
@@ -0,0 +1,41 @@
+import { offscreenStyles } from '../components/offscreen/offscreen-styles.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+
+class AnnounceContainer extends LitElement {
+
+	static get styles() {
+		return [ offscreenStyles ];
+	}
+
+	constructor() {
+		super();
+		this._messages = [];
+	}
+
+	announce(text) {
+		this._messages.push(text);
+		this.requestUpdate();
+	}
+
+	render() {
+		return html`<div aria-live="polite" class="d2l-offscreen">
+			${this._messages.map(message => html`<div>${message}</div>`)}
+		</div>`;
+	}
+
+}
+
+customElements.define('d2l-announce-container', AnnounceContainer);
```

Comment:
Does this actually need to be a custom element? Not sure how much of an advantage it would be to avoid using one altogether, just slightly worried folks are going to see `<d2l-announce-container>` and use it directly.

Reply:
I toyed with doing it each way. My motivation for making it a custom element was two-fold:

1. I was hopeful that I would be able to append it to the `head` (which was still unsuccessful) to avoid being read by virtual cursor
2. To allow for a localized message such as "recent messages" in case it indeed needs to go into body

But yeah, there would be a risk of someone using this arbitrarily.  If we aren't worried about screen reader users discovering the messages without any context, this could easily just be a `div`.

Reply:
Cool, yeah I'm good either way.

Reply:
I spoke with Carin a bunch about this earlier. She confirmed that the last message is announced when navigating the virtual cursor, and while it's a little strange/unexpected, she's learned to ignore it.  She also didn't really recall it ever being raised as an issue, but that if the contents grow and there isn't any context provided, it could become annoying. She acknowledged that there might be some dropped messages due to multiple simultaneous announcements, but she really could say for sure. So with all that in mind, I've updated this PR to maintain the "status quo", mostly in the interest of keeping the region's content to a minimum. We can always change the impl here in the future if necessary.

In my investigation, it turns out that `aria-live` is still very poorly supported across the possible screen reader - browser combinations.  There are certain ones that work well, but there's no single screen reader / implementation that works across all the browsers.  Carin notes some peculiar behavior with them as well.